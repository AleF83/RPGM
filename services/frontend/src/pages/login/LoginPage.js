import { connect } from 'react-redux';
import { compose, lifecycle } from 'recompose';

import LoginPageElement from './LoginPageElement';
import { authProvidersRequest } from './state/authActionCreators';

const mapStateToProps = state => ({
  authProviders: state.auth.authProviders,
});

const mapDispatchToProps = dispatch => ({
  getAuthProviders: () => dispatch(authProvidersRequest()),
});

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps),
  lifecycle({
    componentDidMount() {
      this.props.getAuthProviders();
    },
  }),
);

export default enhance(LoginPageElement);
