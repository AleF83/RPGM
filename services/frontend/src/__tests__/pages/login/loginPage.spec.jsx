/* global test expect */
import React from 'react';
import renderer from 'react-test-renderer';
import LoginPage from '../../../pages/login/LoginPage';

test('Login Page test', () => {
  const component = renderer.create(<LoginPage />);
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
