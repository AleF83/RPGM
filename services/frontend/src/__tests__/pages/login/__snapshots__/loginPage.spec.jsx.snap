// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page test 1`] = `
<div>
  This is Login Page
</div>
`;
